files-csharp/CopyAllContentInDirectory/CopyAllContentInDirectoryTests/CopyAllContentInDirectoryTests.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+
+namespace CopyAllContentInDirectoryTests
+{
+    [TestClass]
+    public class CopyAllContentInDirectoryTests
+    {
+        [TestMethod]
+        public void GivenRunningOnWindows_WhenRunningMain_ThenDirectoryDeepCopyOccurs()
+        {
+            var mockFileSystem = new MockFileSystem();
+
+            // Requires a using statement for System.Runtime.InteropServices;
+            bool isExecutingOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
+            if (!isExecutingOnWindows)
+            {
+                Assert.Inconclusive();
+            }
+
+            Program.SourceDir = "C:\\Source";
+            Program.DestinationDir = "C:\\Destination";
+            Program.FileSystem = mockFileSystem;
+
+            mockFileSystem.AddDirectory("C:\\Source");
+            mockFileSystem.AddDirectory("C:\\Destination");
+            mockFileSystem.AddDirectory("C:\\Source\\ChildDir");
+            mockFileSystem.AddDirectory("C:\\Source\\ChildDir\\ChildOfChildDir");
+
+            mockFileSystem.AddFile("C:\\Source\\ChildDir\\childfile.txt", new MockFileData("childfile"));
+            mockFileSystem.AddFile("C:\\Source\\ChildDir\\ChildOfChildDir\\childofchildfile.txt", new MockFileData("childofchildfile"));
+
+
+            Program.Main();
+
+            Assert.IsTrue(mockFileSystem.AllDirectories.Contains("C:\\Destination\\ChildDir\\ChildOfChildDir"));
+
+            Assert.IsTrue(mockFileSystem.FileExists("C:\\Destination\\ChildDir\\childfile.txt"));
+            Assert.IsTrue(mockFileSystem.FileExists("C:\\Destination\\ChildDir\\ChildOfChildDir\\childofchildfile.txt"));
+        }
+
+        [TestMethod]
+        public void GivenRunningOnLinux_WhenRunningMain_ThenDirectoryDeepCopyOccurs()
+        {
+            var mockFileSystem = new MockFileSystem();
+            bool isExecutingOnLinuxOS = RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+
+            if (!isExecutingOnLinuxOS)
+            {
+                Assert.Inconclusive();
+            }
+
+            Program.SourceDir = "/Source";
+            Program.DestinationDir = "/Destination";
+            Program.FileSystem = mockFileSystem;
+
+            mockFileSystem.AddDirectory("/Source");
+            mockFileSystem.AddDirectory("/Destination");
+            mockFileSystem.AddDirectory("/Source/ChildDir");
+            mockFileSystem.AddDirectory("/Source/ChildDir/ChildOfChildDir");
+
+            mockFileSystem.AddFile("/Source/ChildDir/childfile.txt", new MockFileData("childfile"));
+            mockFileSystem.AddFile("/Source/ChildDir/ChildOfChildDir/childofchildfile.txt", new MockFileData("childofchildfile"));
+
+            Program.Main();
+
+            Assert.IsTrue(mockFileSystem.AllDirectories.Contains("/Destination"));
+            Assert.IsTrue(mockFileSystem.AllDirectories.Contains("/Destination/ChildDir"));
+            Assert.IsTrue(mockFileSystem.AllDirectories.Contains("/Destination/ChildDir/ChildOfChildDir"));
```

Comment:
Same thing as before, just testing for the last one would do.

Reply:
done!